src/transport/SecureSessionMgr.cpp

```diff
@@ -134,8 +134,6 @@ CHIP_ERROR SecureSessionMgr::SendEncryptedMessage(SecureSessionHandle session, E
     PacketHeader packetHeader;
     ReturnErrorOnFailure(packetHeader.DecodeAndConsume(msgBuf));
 
-    VerifyOrReturnError(msgBuf->TotalLength() <= kMaxAppMessageLen + packetHeader.EncodeSizeBytes(), CHIP_ERROR_MESSAGE_TOO_LONG);
```

Comment:
I believe packetHeader is not encrypted and payload header is.

Could you explain why this check is wrong? My logic would be that if we can decode it, the data should be valid here.

Reply:
The check is wrong because it's not checking the right things.

The check we want to make is "application payload is no longer than kMaxAppMessageLen".

The current state on this line is that `msgBuf->TotalLength()` includes the application payload, packet header, and MIC.  All of this is encrypted.  So there's no way to check the size of the application payload here....